test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -4577,6 +4577,42 @@ BOOST_AUTO_TEST_CASE(constructing_enums_from_ints)
 	ABI_CHECK(callContractFunction("test()"), encodeArgs(1));
 }
 
+BOOST_AUTO_TEST_CASE(enum_referencing)
+{
+	char const* sourceCode = R"(
+			interface I {
+				enum Direction { Left, Right }
+			}
+			library L {
+				enum Direction { Left, Right }
+				function f() public pure returns (Direction) {
+					return Direction.Right;
+				}
+				function g() public pure returns (I.Direction) {
+					return I.Direction.Right;
+				}
+			}
+			contract C is I {
+				function f() public pure returns (Direction) {
+					return Direction.Right;
+				}
+				function g() public pure returns (I.Direction) {
+					return I.Direction.Right;
+				}
+				function h() public pure returns (L.Direction) {
+					return L.Direction.Right;
+				}
+			}
+	)";
+	compileAndRun(sourceCode, 0, "C");
```

Comment:
I am not sure library linking is implemented in the test suite?

Reply:
yes, it is automatic

Reply:
Actually I think it is not.